source/resources/examples/typescript/blockchain/ValidatingTransactionWithinBlock.js

```diff
@@ -0,0 +1,63 @@
+"use strict";
+/*
+ *
+ * Copyright 2018-present NEM
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
+    return new (P || (P = Promise))(function (resolve, reject) {
+        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
+        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
+        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
+        step((generator = generator.apply(thisArg, _arguments || [])).next());
+    });
+};
+Object.defineProperty(exports, "__esModule", { value: true });
+/* start block 01 */
+const js_sha3_1 = require("js-sha3");
+const nem2_sdk_1 = require("nem2-sdk");
+const nem2_sdk_openapi_typescript_node_client_1 = require("nem2-sdk-openapi-typescript-node-client");
+const validateTransactionInBlock = (leaf, height, blockHttp) => __awaiter(this, void 0, void 0, function* () {
+    // 2. Obtain HRoot; in Symbol, this is stored in the block header.
+    const HRoot = (yield blockHttp.getBlockByHeight(height).toPromise()).blockTransactionsHash;
+    // 3. Request the merkleProof: H1, H7, H10
+    const merkleProof = (yield blockHttp.getMerkleTransaction(height, leaf).toPromise()).merklePath;
+    // 4. Calculate HRoot'.
+    if (merkleProof.length === 0) {
+        // There is a single item in the tree, so HRoot' = leaf.
+        return leaf.toUpperCase() === HRoot.toUpperCase();
+    }
+    const HRoot0 = merkleProof
+        .reduce((proofHash, pathItem) => {
```

Comment:
Should we replace this with the new BlockService?

Reply:
Not at the moment, the intention of this doc is to remark in this doc **how** the merkle root is  computed from the merkle proof. I'm opening a separate issue to create a guide on how to achieve the same results with the SDK, CLI...